instrumentation-api/instrumentation-api.gradle

```diff
@@ -35,3 +35,11 @@ dependencies {
   testImplementation deps.awaitility
   testImplementation deps.opentelemetrySdkTesting
 }
+
+task createVersionFile {
+  def dir = 'src/main/resources/io/opentelemetry/instrumentation/api'
+  file(dir).mkdirs()
+  new File(projectDir, "${dir}/instrumentation-version.txt").text = "${project.version}"
```

Comment:
does this file get bundled into the instrumentation library that uses the API, somehow? Otherwise, this will only provide the version of the API that the instrumentation library is using, not the version of itself.

Reply:
This file gets bundled into the `opentelemetry-instrumentation-api-<version>.jar`.

All the instrumentation libraries have the same version, which is the same version as the project which is the same version as the API.  The instrumentation itself was naming tracers with `InstrumentationVersion.VERSION`.  The instrumentation will pick this up because the instrumentation depends on this api.

I don't think that instrumentation is versioned independently....at least not yet?

Reply:
It's not, but someone could be using various versions of the library instrumentations, if they aren't keeping their versions in sync with each other (or something is coming in via a dependency). 

So, imagine I'm using library Y, which happens to use library X, along with some otel library instrumentation for X. That library instrumentation might be version 1.5.4 and depends transitively on version 1.5.4 of the instrumentation api.

I'm also using library Z, which happens to use library W, along with otel library instrumentation for W. That library instrumentation might be version 1.6.3 and depends transitively on version 1.6.3 of the instrumentation api.

At runtime, only one instrumentation API library will end up resolved onto the classpath, but both library instrumentation would now report with that version, rather than actual version. This will end up being extremely confusing for debugging issues based on the resulting telemetry.

Does that make sense?

Reply:
Your scenario makes sense, but reads to me like a different situation than was described in #2761.  

Is your expectation that both library X and library W, that were compiled against and depend on different versions of the telemetry, can run happily within the same application?  If so, is there a way for that to even be safe without the libs X and W shading the instrumentation?.

Reply:
If they're using compatible versions of the API (i.e. the same major version) then I think they should definitely be able to work together. Are you thinking there's some reason why they couldn't?

Reply:
I would expect the api to have changed between those versions and I would not expect a compatibility guarantee.

Reply:
Once the API has been marked as stable, there will be compatibility guarantees. Major versions will be backwards compatible with themselves.